cuckoo.cc

```diff
@@ -1,16 +1,19 @@
 #include "./cuckoo.h"
+#include <napi.h>
 
-Napi::FunctionReference Cuckoo::constructor;
+template <size_t bits_per_item>
+Napi::FunctionReference Cuckoo<bits_per_item>::constructor;
```

Comment:
This is a static reference. You can't really do this with a template, and I think this is messing things up a bit. In other terms, I don't think using a template is the right approach. You probably want to call `Init` with another parameter, and store the `bits_per_item` as a value in the C++ class.

Reply:
`cuckoofilter.h` uses templates to set `bits_per_item`. Cuckoo class sets `filter` property using a hard coded int for the template. I'm not sure how to define `filter` in `cuckoo.h` without that int. I was thinking of using a class property and copy & paste the class a bunch of times, with just that int changing. Feels wrong tho.

I think I need to read more Napi docs & C++ language reference. 😅

Reply:
Now I remember why I didn't implement those: _templates_.
I don't really know how to solve this, but it would be awesome to learn